components/api/src/main/java/com/hotels/styx/api/extension/ActiveOrigins.java

```diff
@@ -27,4 +29,8 @@
      * @return a list of connection pools for each active origin
      */
     Iterable<RemoteHost> snapshot();
+
+    String getApplicationId();
```

Comment:
Looks like these two methods are never called. We should leave them out.

Reply:
It looks like they're being used by LoadBalancingGroup.kt

Reply:
That file only has references to those methods because you were forced to implement them when an anonymous subclass of `ActiveOrigins` is created.

Reply:
Yash Vyas: these are being used in the AdaptiveFailoverStrategy implemented in CGP, we might be able to work around it, i’ll have to take another look tomorrow and i can add some details around these then,